packages/shopping/src/sequence.ts

```diff
@@ -33,12 +40,47 @@ export class MySequence implements SequenceHandler {
     try {
       const {request, response} = context;
       const route = this.findRoute(request);
-      await this.authenticateRequest(request);
+
+      //
+      // The authentication action utilizes a strategy resolver to find
+      // an authentication strategy by name, and then it calls
+      // strategy.authenticate(request).
+      //
+      // The strategy resolver throws a non-http error if it cannot
+      // resolve the strategy. When the strategy resolver obtains
+      // a strategy, it calls strategy.authentication(request) which
+      // is expected to return a user profile. If the user profile
+      // is undefined, then it throws a non-http error.
+      //
+      // It is necessary to catch these errors
+      // and rethrow them as http errors (in our REST application example)
+      //
+      // Errors thrown by the strategy implementations are http errors
+      // (in our REST application example). We simply rethrow them.
```

Comment:
@b-admike , I am referring to `packages/shopping/src/application.ts` .
My application is a REST application using http protocol, and so the errors that are thrown by my authentication strategy implementations are http errors. In the future, we can have a set of strategies that handle gRPC protocol requests, and the errors they would throw would be gRPC server errors, and the sequence would have to also catch the protocol-agnostic errors and rethrow them as gRPC errors.

Reply:
Oh okay, I'm mentioning it because I was a bit confused, so others might be too. If you can add what you've explained as a comment to users, that'd be great.

Reply:
@b-admike , this is explained in https://github.com/strongloop/loopback-next/blob/master/packages/authentication/docs/authentication-system.md , and in  in the @loopback/authentication module's mock sequence. I think the comments here explain it well enough.

@jannyHou and I plan to merge the  https://github.com/strongloop/loopback-next/blob/master/packages/authentication/docs files into the main [README.MD](https://github.com/strongloop/loopback-next/blob/master/packages/authentication/README.md)